src/django_components/component_registry.py

```diff
@@ -360,22 +358,27 @@ def unregister(self, name: str) -> None:
         entry = self._registry[name]
         tag = entry.tag
 
-        # Unregister the tag from library if this was the last component using this tag
-        # Unlink component from tag
-        self._tags[tag].remove(name)
+        # Unregister the tag from library.
+        # If this was the last component using this tag, unlink component from tag.
+        if tag in self._tags:
```

Comment:
Changes in this block were simply to handle when the template tag / component names were already unregistered by.

This is now needed, because this method may now be triggered automatically as part of tear down. And we'd run into an index / key error when user explicitly unregisters the component, and then we try to do it again.